src/sampleutil.cpp

```diff
@@ -296,26 +296,31 @@ void SampleUtil::convertS16ToFloat32(CSAMPLE* _RESTRICT pDest, const SAMPLE* _RE
 }
 
 // static
-bool SampleUtil::sumAbsPerChannel(CSAMPLE* pfAbsL, CSAMPLE* pfAbsR,
+SampleUtil::CLIPPING SampleUtil::sumAbsPerChannel(CSAMPLE* pfAbsL, CSAMPLE* pfAbsR,
         const CSAMPLE* pBuffer, int iNumSamples) {
     CSAMPLE fAbsL = CSAMPLE_ZERO;
     CSAMPLE fAbsR = CSAMPLE_ZERO;
-    CSAMPLE clipped = 0;
+    CSAMPLE clippedL = 0;
+    CSAMPLE clippedR = 0;
```

Comment:
Ah, you are right, that destroys vectorizing. I have just tested it with gcc 4.8
You can only do it after the loop like 
```
if (clippedL) {
clipped = static_cast<SampleUtil::CLIP_STATUS>(clipped | SampleUtil::CLIPPING_RIGHT);
}        
```

You can also avoid the ugly casts by overriding the | operator
```
inline CLIP_STATUS operator|(CLIP_STATUS a, CLIP_STATUS b)
{return static_cast<CLIP_STATUS>(static_cast<int>(a) | static_cast<int>(b));}
```